tests/codebuild/run_canarytest.sh

```diff
@@ -49,11 +49,20 @@ echo "Launching canary test for ${COMMIT_SHA}"
 
 # Launch EKS cluster if we need to and define cluster_name,CLUSTER_REGION.
 echo "Launching the cluster"
-readonly cluster_name="sagemaker-k8s-pipeline-"$(date '+%Y-%m-%d-%H-%M-%S')""
 
-# By default eksctl picks random AZ, which time to time leads to  capacity issue.
-# Generally 1a, 1b, 1c are topmost available AZ, hence specifying it explicitly 
-eksctl create cluster "${cluster_name}" --nodes 1 --node-type=c5.xlarge --timeout=40m --region "${CLUSTER_REGION}" --auto-kubeconfig --version ${CLUSTER_VERSION} 
+cluster_name="sagemaker-k8s-canary-"$(date '+%Y-%m-%d-%H-%M-%S')""
+
+if [ -z "${NON_EPHEMERAL_CLUSTER}" ]
+then
+   eksctl_args=( --nodes 1 --node-type=c5.xlarge --timeout=40m --region "${CLUSTER_REGION}" --auto-kubeconfig --version "${CLUSTER_VERSION}" )
+   [ "${CLUSTER_PUBLIC_SUBNETS}" != "" ] && eksctl_args+=( --vpc-public-subnets="${CLUSTER_PUBLIC_SUBNETS}" )
+   [ "${CLUSTER_PRIVATE_SUBNETS}" != "" ] && eksctl_args+=( --vpc-private-subnets="${CLUSTER_PRIVATE_SUBNETS}" )
+
+   eksctl create cluster "${cluster_name}" "${eksctl_args[@]}"
+else
+   cluster_name="non-ephemeral-cluster"
+   aws eks update-kubeconfig --name "${cluster_name}" --region "${CLUSTER_REGION}"
```

Comment:
Does this place the updated kubeconfig in `~/.kube/eksctl/clusters/<name>` or `~/.kube/config`?

Reply:
From the awscli docs:
```
Added new context arn:aws:eks:us-west-2:012345678910:cluster/example to /Users/ericn/.kube/config
```

Reply:
Thanks for looking up for me :) So will this work with line 68?

```
export KUBECONFIG="/root/.kube/eksctl/clusters/${cluster_name}"
```

Reply:
Why would we need to export that env var? What uses that?

Reply:
yeah I am not sure if we need env in second case.